engine/test/integration_tests/dm_many_tables/run.sh

```diff
@@ -27,13 +27,27 @@ function run() {
 		run_sql --quiet "INSERT INTO dm_many_tables.t$i VALUES (9, 90009);"
 	done
 
-	# create job & wait for job finished
+	# create job & wait for job to enter load phase
 	job_id=$(create_job "DM" "$CUR_DIR/conf/job.yaml" "dm_many_tables")
-	# check progress is forwarded gradually, not jump to "finished"
-	exec_with_retry --count 500 "curl \"http://127.0.0.1:10245/api/v1/jobs/$job_id/status\" | tee /dev/stderr | jq -e '.task_status.\"mysql-01\".status.status | .finishedBytes > 0 and .finishedBytes < .totalBytes'"
-	exec_with_retry --count 100 "curl \"http://127.0.0.1:10245/api/v1/jobs/$job_id\" | tee /dev/stderr | jq -e '.state == \"Finished\"'"
-
-	# check data
+	exec_with_retry --count 50 --interval_sec 10 "curl \"http://127.0.0.1:10245/api/v1/jobs/$job_id/status\" | tee /dev/stderr | jq -e '.task_status.\"mysql-01\".status.status | .finishedBytes > 0 and .finishedBytes < .totalBytes'"
```

Comment:
if the interval is 10 seconds, maybe we will miss the timing, this will become a flaky test